java/dataset/src/test/resources/avroschema/user.avsc

```diff
@@ -0,0 +1,26 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+{
```

Comment:
Given the use of only one schema, it might be easier to create this in Java? so readers don't need to look into two places?

Reply:
I think I understand your point here... But my feeling is that the JSON representation of Avro schema is more human-readable than code. e.g. The same schema we create in Java will end up like

```java
public static final Schema AVRO_SCHEMA_USER = Schema.createRecord(
      Arrays.asList(
          new Schema.Field("id",
              Schema.createUnion(Schema.create(Schema.Type.INT), Schema.create(Schema.Type.NULL)),
              null, (Object) null),
          new Schema.Field("name",
              Schema.createUnion(Schema.create(Schema.Type.STRING), Schema.create(Schema.Type.NULL)),
              null, (Object) null)));
```

I think it's a little bit messy but if it's considered better I can change to use this way. I am ok to each.

Reply:
Also there will be a name collision between org.apache.arrow.vector.types.pojo.Schema and org.apache.avro.Schema if we use them in the same file. To avoid writing complicated code due to this we may tend to use a separated Java file to create avro schemas e.g. AvroSchemas.java, which finally makes user look into two places too.

Reply:
I personally agree with @emkornfield  on this one. While harder to read the code than json its more compact and isolated.

Reply:
still think it would be a good idea to inline this in java it would make the tests overall easier to follow because everything is in one place.